Document k8s addon & prometheus metrics

awh opened the pull request:
Fixes weaveworks/weave-kube#30, fixes weaveworks/weave-kube#43, fixes weaveworks/weave-npc#17.

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

awh commented on site/metrics.md:
```diff
@@ -0,0 +1,40 @@
+## Metrics
+
+Two endpoints are exposed; one for the Weave Net router, and, when
+deployed as a Kubernetes Addon, one for the network policy controller:
```
Done.

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

brb commented on site/metrics.md:
```diff
@@ -0,0 +1,146 @@
+      - source_labels: [__meta_kubernetes_pod_container_name,__address__]
+        action: replace
+        target_label: __address__
+        regex: ^weave;(.+)$
```
Could you replace by `regex: ^weave;(.+?)(?::\d+)?$`, because exposing any `containerPort` in the weave-kube manifest would include that port in the `__address__` label?

brb commented on site/metrics.md:
```diff
@@ -0,0 +1,146 @@
+      - source_labels: [__meta_kubernetes_pod_container_name,__address__]
+        action: replace
+        target_label: __address__
+        regex: ^weave-npc;(.+)$
```
Same as above, just `regex: ^weave-npc;(.+?)(?::\d+)?$`

brb replied:
Also, /cc @juliusv for a review

awh replied:
Done.

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

juliusv commented on site/metrics.md:
````diff
@@ -0,0 +1,146 @@
+
+```
+scrape_configs:
+- job_name: 'static-weave'
````
I wouldn't change the job name in the examples based on how the targets are discovered. We'll want to rely on a standard job name in the Cortex UI, so I'd just call both this and the job in the section below `weave` (or `weave-net`).

awh replied:
Done.

brb commented on site/metrics.md:
```diff
@@ -0,0 +1,146 @@
+        action: replace
+        target_label: __address__
+        regex: ^weave-npc;(.+?)(?::\d+)?$
+        replacement: $1:8686
```
@awh Could you append the following relabel:

```
      - source_labels: [__meta_kubernetes_pod_container_name]
        action: replace
        target_label: job
```

awh replied:
Done.

awh commented on site/metrics.md:
```diff
@@ -0,0 +1,146 @@
+remote_write:
+  url: http://frontend.dev.weave.works/api/prom/push
+  basic_auth:
+    password: $WEAVE_CLOUD_TOKEN
```
Good call - done.

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

brb commented on site/metrics.md:
````diff
@@ -0,0 +1,149 @@
+cluster and configure it to discover and scrape the Weave endpoints:
+
+```
+
````
nit-pick: empty line

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

awh commented:
#### Documentation changes:
added [kube-addon](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/kube-addon/)
added [metrics](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/metrics/)
modified [cni-plugin](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/cni-plugin/)
modified [features](https://dev-weavewww.pantheonsite.io/docs/net/kube-npc-docs/features/)

brb commented:
LGTM
